### DISCUSSION TITLE
Don't open script editor when connecting signals to existing methods

### DISCUSSION THREAD

**Discussion by passivestar** [godotengine/godot-proposals#8836] [Editor]
When you create a new method it's great, because you want to implement it, but when you connect to an existing method I can't think of any reason for that to happen. I keep minimizing vscode

IMO should just not happen at all for existing methods, but if somebody can think of a reason for that to happen then maybe could at least be configurable:

![1](https://github.com/godotengine/godot-proposals/assets/60579014/6f3a87a1-373d-4744-8909-bcbec2ddebf5)

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot-proposals/issues/8351

Almost the same idea with the difference to not switch when the method exists, as opposed to at all

**Comment by AThousandShips** [comment]
I'd say, from the conversation on the other issue, that this should be a setting, not absolute

**Reply by WeaverSong** [reply]
Maybe a three-way setting? Always open, open only on creation, and never open? I can certainly see myself using the middle option most of the time, and from the other conversation it seems that is the current intention.
